### PR TITLE
FIX #10309

### LINKED ISSUE #10309: Bug: Openldap password not correctly set
# Bug
New created member can't authenticate from LDAP

## Environment
- **Version**: Dolibarr 8.0.4
- **OS**: Debian 9
- **Web server**: Apache2
- **PHP**: PHP7
- **Database**: Mysql

## Use case
I have an OpenLDAP for authenticate members with other apps.
After created a new user and test to bind to LDAP I get "invalid credential"
I've test directly LDAP bind, with "Apache directory studio" and test password function, the password is definively not set correctly by dolibarr.

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

File: htdocs/core/lib/security.lib.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ function dol_hash($chain, $type = '0')
 	}
 
 	// Salt value
-	if (! empty($conf->global->MAIN_SECURITY_SALT)) $chain=$conf->global->MAIN_SECURITY_SALT.$chain;
+	if (! empty($conf->global->MAIN_SECURITY_SALT) && ( ! $type == '4' || ! $type == 'md5openldap')) $chain=$conf->global->MAIN_SECURITY_SALT.$chain;
```
Comment (eldy) [inline htdocs/core/lib/security.lib.php]: ! $type == '4'  means (! $type) == '4'
I don't think this is want you want to do.

**Comment by hregis** [comment]
@eldy 
I had made a copy paste of what had been proposed and I forgot to modify it.
I corrected.

File: htdocs/core/lib/security.lib.php [hunk 2]
```diff
@@ -122,7 +122,7 @@ function dol_hash($chain, $type = '0')
 	}
 
 	// Salt value
-	if (! empty($conf->global->MAIN_SECURITY_SALT)) $chain=$conf->global->MAIN_SECURITY_SALT.$chain;
+	if (! empty($conf->global->MAIN_SECURITY_SALT) && ($type != '4' || $type !== 'md5openldap')) $chain = $conf->global->MAIN_SECURITY_SALT.$chain;
```
Comment (eldy) [inline htdocs/core/lib/security.lib.php]: This test && ($type != '4' || $type !== 'md5openldap') seems useless as it is always true

Comment (hregis) [reply htdocs/core/lib/security.lib.php]: @eldy arf oui j'ai corrigé... merci

**Pull request merged**